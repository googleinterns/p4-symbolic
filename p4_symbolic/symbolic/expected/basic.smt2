; 
(set-info :status unknown)
(declare-fun standard_metadata.ingress_port () (_ BitVec 9))
(declare-fun ipv4.dstAddr () (_ BitVec 32))
(declare-fun ipv4.$valid$ () Bool)
(assert
 (let (($x63 (= standard_metadata.ingress_port (_ bv1 9))))
 (or (or false (= standard_metadata.ingress_port (_ bv0 9))) $x63)))
(assert
 (let (($x89 (and true (= ((_ extract 31 16) ipv4.dstAddr) ((_ extract 31 16) (_ bv336855040 32))))))
 (let (($x109 (and true (= ((_ extract 31 16) ipv4.dstAddr) ((_ extract 31 16) (_ bv168427520 32))))))
 (let ((?x129 (ite ipv4.$valid$ (ite $x109 0 (ite $x89 1 (- 1))) (- 1))))
 (let (($x128 (ite ipv4.$valid$ (or (or true $x89) $x109) false)))
 (let (($x127 (ite ipv4.$valid$ (or false (ite $x109 false (ite $x89 false true))) false)))
 (and (and (not $x127) $x128) (= ?x129 (- 1)))))))))
(check-sat)

; 
(set-info :status unknown)
(declare-fun standard_metadata.ingress_port () (_ BitVec 9))
(declare-fun ipv4.dstAddr () (_ BitVec 32))
(declare-fun ipv4.$valid$ () Bool)
(assert
 (let (($x63 (= standard_metadata.ingress_port (_ bv1 9))))
 (or (or false (= standard_metadata.ingress_port (_ bv0 9))) $x63)))
(assert
 (let (($x89 (and true (= ((_ extract 31 16) ipv4.dstAddr) ((_ extract 31 16) (_ bv336855040 32))))))
 (let (($x109 (and true (= ((_ extract 31 16) ipv4.dstAddr) ((_ extract 31 16) (_ bv168427520 32))))))
 (let ((?x129 (ite ipv4.$valid$ (ite $x109 0 (ite $x89 1 (- 1))) (- 1))))
 (let (($x128 (ite ipv4.$valid$ (or (or true $x89) $x109) false)))
 (let (($x127 (ite ipv4.$valid$ (or false (ite $x109 false (ite $x89 false true))) false)))
 (let (($x141 (and (not $x127) $x128)))
 (and $x141 (= ?x129 0)))))))))
(check-sat)

; 
(set-info :status unknown)
(declare-fun standard_metadata.ingress_port () (_ BitVec 9))
(declare-fun ipv4.dstAddr () (_ BitVec 32))
(declare-fun ipv4.$valid$ () Bool)
(assert
 (let (($x63 (= standard_metadata.ingress_port (_ bv1 9))))
 (or (or false (= standard_metadata.ingress_port (_ bv0 9))) $x63)))
(assert
 (let (($x89 (and true (= ((_ extract 31 16) ipv4.dstAddr) ((_ extract 31 16) (_ bv336855040 32))))))
 (let (($x109 (and true (= ((_ extract 31 16) ipv4.dstAddr) ((_ extract 31 16) (_ bv168427520 32))))))
 (let ((?x129 (ite ipv4.$valid$ (ite $x109 0 (ite $x89 1 (- 1))) (- 1))))
 (let (($x128 (ite ipv4.$valid$ (or (or true $x89) $x109) false)))
 (let (($x127 (ite ipv4.$valid$ (or false (ite $x109 false (ite $x89 false true))) false)))
 (and (and (not $x127) $x128) (= ?x129 1))))))))
(check-sat)

