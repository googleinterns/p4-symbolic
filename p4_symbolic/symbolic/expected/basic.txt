Finding packet for table MyIngress.ipv4_lpm and row -1
Cannot find solution!

Finding packet for table MyIngress.ipv4_lpm and row 0
	standard_metadata.ingress_port = #b000000000
	standard_metadata.egress_spec = #b000000000
	ipv4.dstAddr = #x0a0a0000
	ethernet.dstAddr = #x000000000000

Finding packet for table MyIngress.ipv4_lpm and row 1
	standard_metadata.ingress_port = #b000000000
	standard_metadata.egress_spec = #b000000001
	ipv4.dstAddr = #x14140000
	ethernet.dstAddr = #x160000000016

